pkg/services/datasources/service/store.go

```diff
@@ -399,3 +405,10 @@ func generateNewDatasourceUid(sess *db.Session, orgId int64) (string, error) {
 }
 
 var generateNewUid func() string = util.GenerateShortUID
+
+func logDeprecatedInvalidDsUid(logger log.Logger, uid, name string) {
+	logger.Error(
+		"invalid datasource uid, invalid uids are deprecated and this operation will fail in future versions of Grafana",
+		"uid", uid, "name", name,
+	)
```

Comment:
Nit. Maybe some more details what makes a uid valid might be a good idea? Maybe too much text in a log message?

```suggestion
	logger.Warn(
		"Invalid datasource uid. The use of invalid uids are deprecated and this operation will fail in a future versions of Grafana. A valid uid is a combination of a-z, A-Z, 0-9 (alphanumeric), - (dash) and _ (underscore) characters, maximum length 40. ",
		"uid", uid, "name", name,
	)
```

As an alternative or in addition maybe we should add something similar as https://grafana.com/docs/grafana/latest/developers/http_api/dashboard/#identifier-id-vs-unique-identifier-uid to https://grafana.com/docs/grafana/latest/developers/http_api/data_source/

Reply:
I feel maybe it's too long for a log line but at the same time it's great to give more info to the user on how to fix the issue as well. If we include a URL to the docs it will become very long anyways.

So I think the best solution is applying your code suggestion.

@andresmgot @wbrowne wdyt?

Reply:
I agree on having an actionable item and no link. If we want to make it a bit shorter I suggest: `Deprecation notice: Invalid datasource uid. A valid uid is a combination of a-z, A-Z, 0-9 (alphanumeric), - (dash) and _ (underscore) characters, maximum length 40.`

Reply:
I was thinking, what if we use a regex instead?

```
invalid datasource uid. The use of invalid uids is deprecated and this operation will fail in a future version of Grafana. A valid uid matches the regex ^[a-zA-Z0-9\-\_]{1,40}$ logger=datasources uid="Oracle 11 (Integration) new" name="Oracle 11 (Integration) new"
```

It is shorter but do we expect Grafana operators to understand regexes?

Reply:
I prefer the human readable version :shrug: